specification/keyvault/data-plane/Security.KeyVault.Secrets/models.tsp

```diff
@@ -0,0 +1,402 @@
+
+  /**
+   * Denotes a vault state in which deletion is recoverable without the possibility
+   * for immediate and permanent deletion (i.e. purge when 7<=
+   * SoftDeleteRetentionInDays < 90).This level guarantees the recoverability of the
+   * deleted entity during the retention interval and while the subscription is
+   * still available.
+   */
+  CustomizedRecoverable: "CustomizedRecoverable",
+
+  /**
+   * Denotes a vault and subscription state in which deletion is recoverable,
+   * immediate and permanent deletion (i.e. purge) is not permitted, and in which
+   * the subscription itself cannot be permanently canceled when 7<=
+   * SoftDeleteRetentionInDays < 90. This level guarantees the recoverability of the
+   * deleted entity during the retention interval, and also reflects the fact that
+   * the subscription itself cannot be cancelled.
+   */
+  CustomizedRecoverable_ProtectedSubscription: "CustomizedRecoverable+ProtectedSubscription",
+}
+
+/**
+ * The secret set parameters.
+ */
+model SecretSetParameters {
+  /**
+   * The value of the secret.
+   */
+  value: string;
+
+  /**
+   * Application specific metadata in the form of key-value pairs.
+   */
+  tags?: Record<string>;
+
+  /**
+   * Type of the secret value such as a password.
+   */
+  contentType?: string;
+
+  /**
+   * The secret management attributes.
+   */
+  @encodedName("application/json", "attributes")
+  secretAttributes?: SecretAttributes;
+}
+
+/**
+ * The secret management attributes.
+ */
+model SecretAttributes {
+  ...Attributes;
+
+  /**
+   * softDelete data retention days. Value should be >=7 and <=90 when softDelete
+   * enabled, otherwise 0.
+   */
+  @visibility("read")
+  recoverableDays?: int32;
+
+  /**
+   * Reflects the deletion recovery level currently in effect for secrets in the
+   * current vault. If it contains 'Purgeable', the secret can be permanently
+   * deleted by a privileged user; otherwise, only the system can purge the secret,
+   * at the end of the retention interval.
+   */
+  @visibility("read")
+  recoveryLevel?: DeletionRecoveryLevel;
+}
+
+/**
+ * The object attributes managed by the KeyVault service.
+ */
+model Attributes {
+  /**
+   * Determines whether the object is enabled.
+   */
+  enabled?: boolean;
+
+  /**
+   * Not before date in UTC.
+   */
+  @encodedName("application/json", "nbf")
+  @encode("unixTimestamp", int32)
+  notBefore?: utcDateTime;
+
+  /**
+   * Expiry date in UTC.
+   */
+  @encodedName("application/json", "exp")
+  @encode("unixTimestamp", int32)
+  expires?: utcDateTime;
+
+  /**
+   * Creation time in UTC.
+   */
+  @visibility("read")
+  @encode("unixTimestamp", int32)
+  created?: utcDateTime;
+
+  /**
+   * Last updated time in UTC.
+   */
+  @visibility("read")
+  @encode("unixTimestamp", int32)
+  updated?: utcDateTime;
+}
+
+/**
+ * A secret consisting of a value, id and its attributes.
+ */
+model SecretBundle {
```

Comment:
Is there going to be a client TSP? This should be `KeyVaultSecret`, and other models for keys, certs should be `KeyVaultKey` and `KeyVaultCertificate`. The architects didn't want one-word model types that are generic like "Secret", "Key", "Certificate".

Reply:
This model naming is intentional, to match the original spec. The goal was to create no breaking changes after generation, which is shown in the Python integration: https://github.com/Azure/azure-sdk-for-python/pull/36901

Reply:
Understood, which is why I mentioned a client TSP. :wink: